src/neo/SmartContract/Native/GasToken.cs

```diff
@@ -33,5 +36,14 @@ internal override async ContractTask OnPersist(ApplicationEngine engine)
             UInt160 primary = Contract.CreateSignatureRedeemScript(validators[engine.PersistingBlock.PrimaryIndex]).ToScriptHash();
             await Mint(engine, primary, totalNetworkFee, false);
         }
+
+        [ContractMethod(RequiredCallFlags = CallFlags.States | CallFlags.AllowNotify)]
```

Comment:
`Fee` for the syscall? otherwise the syscall is free, burn should be done because you will consume it later.